src/components/textfieldControl/js/Textfield.js

```diff
@@ -0,0 +1,138 @@
+
+    /*************
+     * Textfield *
+     *************/
+
+    /**
+     * A component for controlling a textfield and handling data binding.
+     * Typically this will be used in conjunction with a UI control widget such as
+     * button steppers or slider.
+     */
+    fluid.defaults("fluid.textfield", {
+        gradeNames: ["fluid.viewComponent"],
+        ariaOptions: {
+            // Specified by implementor
+            // ID of an external label to refer to with aria-labelledby
+            // attribute
+            // "aria-labelledby": ""
+        },
+        strings: {
+            // Specified by implementor
+            // text of label to apply to both textfield and slider input
+            // via aria-label attribute
+            // "aria-label": ""
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            }
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            },
+            "onCreate.initTextfieldAttributes": {
+                "this": "{that}.container",
+                method: "attr",
+                args: [{
+                    "aria-labelledby": "{that}.options.ariaOptions.aria-labelledby",
+                    "aria-label": "{that}.options.strings.aria-label"
+                }]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "value",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    /**
+     * Sets the model value only if the new value is a valid number, and will reset the textfield to the current model
+     * value otherwise.
+     *
+     * @param {Object} that - the component
+     * @param {Number} value - the new numerical entry
+     * @param {String} path - the path into the model for which the value should be set
+     */
+    fluid.textfield.setModelRestrictToNumbers = function (that, value, path) {
+        var isNumber = !isNaN(Number(value));
+        if (isNumber) {
+            that.applier.change(path, value);
+        }
+
+        // Set the textfield to the latest valid entry.
+        // This handles both the cases where an invalid entry was provided, as well as cases where a valid number is
+        // rounded. In the case of rounded numbers this ensures that entering a number that rounds to the current
+        // set value, doesn't leave the textfield with the unrounded number present.
+        that.container.val(that.model.value);
+    };
+
+    /******************************
+     * TextField Range Controller *
+     ******************************/
+
+    /**
+     * Range Controller is intended to be used as a subcomponent of a fluid.textfield component. It will limit the input
```

Comment:
This statement `Range Controller is intended to be used as a subcomponent of a fluid.textfield component.` is wrong. The code shows range controller uses `fluid.textfield` as a base grade instead of being its subcomponent.

I'm not sure if it's necessary to have 2 versions of text field: one is a pure text field and the other is the pure plus the value verification. All our use cases so far with slider or stepper only need the latter. If this implementation is to better modularize the component, I'm ok with it. But I would suggest to rename `fluid.textfield.rangeController` as `fluid.textfiled` because this would be the one used most often and a simple name would be easier to use and reduce confusion.